text/0097-log-data-model.md

```diff
@@ -0,0 +1,1340 @@
+15            |WARN3
+16            |WARN4
+17            |ERROR
+18            |ERROR2
+19            |ERROR3
+20            |ERROR4
+21            |FATAL
+22            |FATAL2
+23            |FATAL3
+24            |FATAL4
+
+When an individual log record is displayed it is recommended to show both
+`SeverityText` and `SeverityNumber` values. A recommended combined string in
+this case begins with the short name followed by `SeverityText` in parenthesis.
+
+For example "Informational" Syslog record will be displayed as **INFO
+(Informational)**. When for a particular log record the `SeverityNumber` is
+defined but the `SeverityText` is missing it is recommended to only show the
+short name, e.g. **INFO**.
+
+When drop down lists (or other UI elements that are intended to represent the
+possible set of values) are used for representing the severity it is preferable
+to display the short name in such UI elements.
+
+For example a dropdown list of severities that allows filtering log records by
+severities is likely to be more usable if it contains the short names of
+`SeverityNumber` (and thus has a limited upper bound of elements) compared to a
+dropdown list, which lists all distinct `SeverityText` values that are known to
+the system (which can be a large number of elements, often differing only in
+capitalization or abbreviated, e.g. "Info" vs "Information").
+
+#### Comparing Severity
+
+In the contexts where severity participates in less-than / greater-than
+comparisons `SeverityNumber` field should be used. `SeverityNumber` can be
+compared to another `SeverityNumber` or to numbers in the 1..24 range (or to the
+corresponding short names).
+
+When severity is used in equality or inequality comparisons (for example in
+filters in the UIs) the recommendation is to attempt to use both `SeverityText`
+and short name of `SeverityNumber` to perform matches. For example if we have a
+record with `SeverityText` field equal to "Informational" and `SeverityNumber`
+field equal to INFO then it may be preferable from the user experience
+perspective to ensure that **severity="Informational"** and **severity="INFO"**
+conditions both to are TRUE for that record.
+
+### Field: `ShortName`
+
+Type: string.
+
+Description: Short event identifier that does not contain varying parts.
+`ShortName` describes what happened (e.g. "ProcessStarted"). Recommended to be
+no longer than 50 characters. Optional. Not guaranteed to be unique in any way.
+Typically used for filtering and grouping purposes in backends.
+
+### Field: `Body`
+
+Type: any.
+
+Description: A value containing the body of the log record (see the description
+of `any` type above). Can be for example a human-readable string message
+(including multi-line) describing the event in a free form or it can be a
+structured data composed of arrays and maps of other values. Can vary for each
+occurrence of the event coming from the same source.
+
+### Field: `Resource`
+
+Type: key/value pair list.
```

Comment:
Should the name be then "Resources" (plural). Before reading the explanation I though the type for this would be a string. It seems it is plural in the OT spec: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#resources

Reply:
It is a singular `Resource` per record. We cannot have multiple resources per log record. The same applies to traces and metrics.

Reply:
Oh, I missed the part in the link I posted that the title is "Resources" but the field is singular. I did not have a chance to comment on that earlier. IMHO, naming lists plural is less ambiguous. There are many use cases a record can be associated with more than one resource, e.g. in k8s, pod, node, cluster, etc. However, to be consistent with the metrics/traces model, it makes sense to keep is as 'Singular' as far as this pr concerned.

Reply:
>There are many use cases a record can be associated with more than one resource

That's a very good point and it has been part of a discussion we have outside of logging SIG about the Resource. For now I will keep this consistent with the rest of OpenTelemetry spec, but this is subject to change if Resource is revised.